core/src/main/scala/akka/persistence/cassandra/query/AllPersistenceIdsStage.scala

```diff
@@ -45,45 +45,45 @@ import scala.concurrent.ExecutionContextExecutor
   def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
     new TimerGraphStageLogic(shape) with StageLogging with OutHandler {
 
-      implicit def executionContext: ExecutionContextExecutor = materializer.executionContext
-
       private var queryInProgress = false
       private var knownPersistenceIds: Set[String] = Set.empty
-      private var maybeResultSet = Option.empty[ResultSet]
+      private var maybeResultSet = Option.empty[AsyncResultSet]
       private var buffer: Queue[String] = Queue.empty[String]
 
-      private val queryCallback: AsyncCallback[ResultSet] =
-        getAsyncCallback[ResultSet] { rs =>
+      private val queryCallback: AsyncCallback[AsyncResultSet] =
+        getAsyncCallback[AsyncResultSet] { rs =>
           queryInProgress = false
           maybeResultSet = Some(rs)
-          val available = rs.getAvailableWithoutFetching
-          for (_ <- 1 to available) {
-            val s = rs.one().getString("persistence_id")
+          rs.currentPage().forEach { row =>
+            val s = row.getString("persistence_id")
             if (!knownPersistenceIds.contains(s)) {
               buffer = buffer.enqueue(s)
               knownPersistenceIds += s
             }
           }
           flush()
-          if (refreshInterval.isEmpty && buffer.isEmpty && rs.isExhausted) {
+          if (refreshInterval.isEmpty && buffer.isEmpty && rs.hasMorePages) {
```

Comment:
should it be `!rs.hasMorePages` to correspond to `rs.isExhausted`?

Reply:
Yep, i switched this to use the common `isExhausted`